cpp/examples/minimal_abm.cpp

```diff
@@ -0,0 +1,158 @@
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+#include "abm/abm.h"
+#include "abm/household.h"
+#include <cstdio>
+
+int main()
+{
+    // Set global infection parameters (similar to infection parameters in SECIR model) and initialize the world
+    mio::abm::GlobalInfectionParameters infection_params;
+
+    // Set same infection parameter for all age groups. For example, the incubation period is 4 days. 
+    infection_params.get<mio::abm::IncubationPeriod>() = 4.;
+    
+    // Create the world with infection parameters.
+    auto world = mio::abm::World(infection_params);
+
+    // There are 3 households for each hosehold group.
+    int n_households = 3;
+
+    // For more than 1 family households we need families. These are parents and children and randoms (which are distributed like the data we have for these households).
+    auto child = mio::abm::HouseholdMember(); // A child is 50/50% 0-4 or 5-14.
+    child.set_age_weight(mio::abm::AgeGroup::Age0to4, 1);
+    child.set_age_weight(mio::abm::AgeGroup::Age5to14, 1);
+
+    auto parent = mio::abm::HouseholdMember(); // A parent is 50/50% 15-34, 35-59 or 60-79.
+    parent.set_age_weight(mio::abm::AgeGroup::Age15to34, 1);
+    parent.set_age_weight(mio::abm::AgeGroup::Age35to59, 1);
+    
+    // Two-person household with one parent and one child. 
+    auto twoPersonHousehold_group = mio::abm::HouseholdGroup();
+    for (int counter = 0; counter < n_households; counter++) {
+        auto twoPersonHousehold_full = mio::abm::Household();
+        twoPersonHousehold_full.add_members(child, 1);
+        twoPersonHousehold_full.add_members(parent, 1);
+        twoPersonHousehold_group.add_households(twoPersonHousehold_full, 2);  
+    }
+    add_household_group_to_world(world, twoPersonHousehold_group);
+
+    // Three-person household with two parent and one child. 
+    auto threePersonHousehold_group = mio::abm::HouseholdGroup();
+    for (int counter = 0; counter < n_households; counter++) {
+        auto threePersonHousehold_full = mio::abm::Household();
+        threePersonHousehold_full.add_members(child, 1);
+        threePersonHousehold_full.add_members(parent, 2);
+        threePersonHousehold_group.add_households(threePersonHousehold_full, 2);
+    }
+    add_household_group_to_world(world, threePersonHousehold_group);
+
+    // Assumed percentage of infection state at the beginning of the simulation.
+    double exposed_pct = 0.01, infected_pct = 0.08, carrier_pct = 0.05, recovered_pct = 0.01;
+    // Assign an infection state to each person.
+    // The infection states are chosen randomly. They are distributed according to the probabilites set in the example.
+    auto persons = world.get_persons();
+    double susceptible          = 1 - exposed_pct - infected_pct - carrier_pct - recovered_pct;
+    std::vector<double> weights = {susceptible, exposed_pct, carrier_pct, infected_pct / 2,
```

Comment:
This initialization is prone to errors, as it assumes the number and position of infection states to be fixed. There had been an error discovered in: https://github.com/DLR-SC/memilio/pull/288

We should use a future-safe approach by using the states enum.

Reply:
The agents are now assigned a random enum state.